### PR TITLE
Add headers for client type and client version

### PR DESCRIPTION
**Pull request by Hinton**
## Type of change

- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

Adds two new headers included in each request. I decided to not re-use the `DeviceType`, since it's more granular and we throw away the client type in `getDeviceString`.

`Bitwarden-Client-Name`: `[web, browser, desktop, cli, connector, mobile]`
`Bitwarden-Client-Version`: `1.41.0`

This will allow us to get an insight into which versions are currently in use.

## Code changes

- **common/src/abstractions/platformUtils.service.ts:** Added new method, `getClientType`.
- **common/src/enums/clientType.ts:** New enum with the client types.

## Testing requirements

Should be covered in general regression testing.

## Before you submit

- [x] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

**Comment by Hinton** [comment]
@joseph-flinn Are there any devops related changes required for this?

File: common/src/services/api.service.ts [hunk 1]
```diff
@@ -225,7 +226,7 @@ export class ApiService implements ApiServiceAbstraction {
     const identityToken =
       request instanceof ApiTokenRequest
         ? request.toIdentityToken()
-        : request.toIdentityToken(this.platformUtilsService.identityClientId);
+        : request.toIdentityToken(this.platformUtilsService.getClientType());
```
Comment (MGibson1) [inline common/src/services/api.service.ts:229]: Update `toIdentityToken` to expect either values in `ClienType` enum or a `ClientType` enum itself

**Review by MGibson1 — CHANGES_REQUESTED** [review]

File: common/src/abstractions/platformUtils.service.ts [hunk 2]
```diff
@@ -6,9 +7,9 @@ interface ToastOptions {
 }
 
 export abstract class PlatformUtilsService {
-  identityClientId: string;
   getDevice: () => DeviceType;
   getDeviceString: () => string;
+  getClientType: () => ClientType;
```
[the remark this answers is not in the document]
Comment (Hinton) [reply common/src/abstractions/platformUtils.service.ts:12]: It already is. If it's not implemented you get an error. (Same as the other)

[the remark this answers is not in the document]
Comment (MGibson1) [reply common/src/abstractions/platformUtils.service.ts:12]: my mistake, I thought it had to be explicitly abstract

**Comment by joseph-flinn** [comment]
@Hinton After a quick review of the changes, I don't think there are any updates to the headers. However, it might be best to make a note on the Jira card for the regression testing phase and to triple check by testing it with the QA cloud environment.

**Comment by MGibson1** [comment]
> After a quick review of the changes, I don't think there are any updates to the headers. However, it might be best to make a note on the Jira card for the regression testing phase and to triple check by testing it with the QA cloud environment.

new headers are added here: https://github.com/bitwarden/jslib/blob/feature/app-client-header/common/src/services/api.service.ts#L2214-L2219

**Review by MGibson1 — APPROVED** [review]

**Pull request merged**